features/polyExplorer/src/components/entityList/entityList.jsx

```diff
@@ -58,19 +17,26 @@ function getStartGroups(entityGroups) {
     return keys.pop();
 }
 
-const EntityList = () => {
-    const { entities, globalData, activeFilters, handleRemoveFilter } =
-        useContext(ExplorerContext);
-    const onRemoveFilter = handleRemoveFilter;
-    const filteredEntities = activeFilters.apply(entities);
-    const entityGroups = groupEntities(filteredEntities);
+export default function EntityList({ entityGroups }) {
     const allKeys = Object.keys(entityGroups);
     const [loadedEntities, setLoadedEntities] = useState({});
     const [toLoadKeys, setToLoadKeys] = useState(allKeys);
     const [hasMore, setHasMore] = useState(true);
     const listRef = useRef();
 
-    const handleLoadMoreData = () => {
+    function loadEntities() {
+        const loadedEntities = {};
+        const startGroups = getStartGroups(entityGroups);
+        for (let i = 0; i <= allKeys.indexOf(startGroups); i++) {
+            loadedEntities[allKeys[i]] = entityGroups[allKeys[i]];
+        }
+        setLoadedEntities(loadedEntities);
+        const toLoadKeys = [...allKeys];
```

Comment:
This command and the two that follow could be a single one, and avoid an extra variable. If that's a bit too much, simply use filter on [..allKeys]

Reply:
I think it's because splice modifies the actual array it's called on. 
I think what you could do is 
`setToLoadKeys(allKeys.**slice**(0, allKeys.indexOf(startGroups)))`

Reply:
No that doesnt work. Splice modifies the array its called on so we want to actually cut that out and for that we need some new throwaway array

Reply:
Yeah true, also more efficient, will add it :D (That said, I tried to touch the code I'm moving around as little as possible - it's a refactoring, not a full on rewrite.)

Reply:
Since I did a slightly bigger refactoring now, I noticed the `indexOf` call can be avoided anyway, if we just return the _number_ of groups to load rather than the _name_ of the last group to load. Simplified everything a bit, including this bit.